.github/workflows/build-test.yaml

```diff
@@ -0,0 +1,61 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.9"
+          cache: "pip"
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          # TODO(markblee): Remove gcp,vertexai_tensorboard from CI. (needed by pytype)
+          pip install '.[dev,gcp,vertexai_tensorboard]'
+      - name: Run pre-commit
+        run: pre-commit run --all-files
+      - name: Run pytype
+        run: pytype -j auto .
+
+  set-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: define pytest testing matrix
+        id: set-matrix
+        run: |
+          echo "matrix=$(python3 .github/workflows/set_matrix.py)" >> $GITHUB_OUTPUT
+
+  build-and-test:
```

Comment:
Is there a way to set the name here to something nicer than the list of files? I'm imagining `build-and-test (1/10)` and then first step can be to echo list of files to output for debugging later?